code/game/objects/items/weapons/lighters.dm

```diff
@@ -118,19 +124,22 @@
 	return
 
 /obj/item/lighter/update_icon_state()
-	icon_state = "[initial(icon_state)][lit ? "-on" : ""]"
+	icon_state = "[base_icon_state ? "[base_icon_state]" : "[initial(icon_state)]"][lighter_color ? "-[lighter_color]" : ""][lit ? "-on" : ""]"
 	return ..()
 
 /obj/item/lighter/update_overlays()
```

Comment:
While you're here, changing the item_state isnt really changing the overlay. This would be better to include in the update_icon_state proc
```suggestion
```